sky/packages/sky/lib/widgets/framework.dart

```diff
@@ -369,6 +369,17 @@ abstract class Widget {
     return '$runtimeType($key; hashCode=$hashCode)';
   }
 
+  Point globalToLocal(Point point) {
+    assert(mounted);
+    assert(root is RenderBox);
+    return (root as RenderBox).globalToLocal(point);
+  }
+
+  Point localToGlobal(Point point) {
+    assert(mounted);
+    assert(root is RenderBox);
+    return (root as RenderBox).localToGlobal(point);
+  }
```

Comment:
These functions are a little dodgy because we don't know if the RenderBox's layout and such are up-to-date.  Let's leave them out of the first patch and see how far we get without them.